libs/core/src/lib/input-group/input-group-directives.ts

```diff
@@ -5,30 +5,42 @@ import {
     HostBinding,
     Input,
     OnChanges,
+    OnDestroy,
     OnInit,
+    Optional,
     Renderer2
 } from '@angular/core';
 import { InputGroupPlacement } from './input-group.component';
 import { FormStates } from '../form/form-control/form-states';
-import { applyCssClass, CssClassBuilder } from '../utils/public_api';
+import { applyCssClass, ContentDensityService, CssClassBuilder } from '../utils/public_api';
+import { Subscription } from 'rxjs';
 
 @Directive({
     // tslint:disable-next-line:directive-selector
     selector: '[fd-input-group-input]'
 })
-export class InputGroupInputDirective implements CssClassBuilder, OnInit, OnChanges {
+export class InputGroupInputDirective implements CssClassBuilder, OnInit, OnChanges, OnDestroy {
     /** user's custom classes */
     @Input()
     class: string;
 
     @Input()
-    compact = false;
+    compact: boolean = null;
```

Comment:
please use undefined and a proper typings, ng 12 will have strict enabled by default, so we should comply
```suggestion
    compact?: boolean;
```